fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -306,6 +300,10 @@ private Type getCmpType() {
             return Type.LARGEINT;
         }
 
+        if ((t1.isNumericType() && t2.isStringType()) || (t1.isStringType() && t2.isNumericType())) {
+            return t1.isNumericType() ? Type.fromPrimitiveType(t1) : Type.fromPrimitiveType(t2);
+        }
```

Comment:
what about if "LargeIntType"(16byte) compare "DoubelType"(8byte) ? 
what about if "DecimalValuetType"(16byte) compare "DoubelType"(8byte)  ? 
what about if "LargeIntType" compare "DecimalType" ? 
what about if "DecimalType" compare "DecimalType" ?

Reply:
all the above will both cast to a middle type `DoubleType` and to compare, I just handle the situation that compare `Numerical` and  `StringType`

Reply:
At least equivalent comparison (decimaltype to decimaltype) does not need to be converted to double type, You can fix it by the way.

Reply:
if they are both DecimalType, the function will return DecimalType, it's already checked in the function.

Reply:
Can you explain in detail if both input parameters are decimal type.， why this funciton return decimal type？

Reply:
you can get it in the following code
```
if ((t1 == PrimitiveType.BIGINT || t1 == PrimitiveType.DECIMAL)
                && (t2 == PrimitiveType.BIGINT || t2 == PrimitiveType.DECIMAL)) {
        return Type.DECIMAL;
}
```